### PR TITLE
Update test wording to avoid confusion with "activation" terminology

### LINKED ISSUE #169: Rewording test case to avoid confusion with "activation" term
**Problem description**

Update following test scenario to avoid confusion with "activation" terminology

```
@retrieve_sim_swap_date_4_sim_not_activated
  Scenario: Retrieves SIM swap date for a non-activated sim
    Given a valid phone number identified by the token or provided in the request body
    And the sim for that device has never been connected to the Operator's network
    When the request "retrieveSimSwapDate" is sent
    Then the response status code is 200
    And the response property "$.latestSimChange" is null
```

with

```
@retrieve_sim_swap_date_4_sim_never_connected
  Scenario: Retrieves SIM swap date for a sim that has never been connected to the network
    Given a valid phone number identified by the token or provided in the request body
    And the sim for that phone number has never been connected to the Operator's network
    When the request "retrieveSimSwapDate" is sent
    Then the response status code is 200
    And the response property "$.latestSimChange" is null
```

### PR DESCRIPTION
**Pull request by fernandopradocabrillo**
#### What type of PR is this?

Add one of the following kinds:
* correction
* tests

#### What this PR does / why we need it:

Updates test @retrieve_sim_swap_date_4_*  wording

#### Which issue(s) this PR fixes:

Fixes #169

### CODE REVIEW

File: code/Test_definitions/sim-swap-retrieveDate.feature [hunk 1]
```diff
@@ -45,10 +45,10 @@ Feature: CAMARA SIM Swap API, 1.0.0 - Operation retrieveSimSwapDate
     And the response property "$.latestSimChange" contains the sim's activation timestamp
 
   # This test applies if the operator allows to do the request for a sim that has never been connected to the network
-  @retrieve_sim_swap_date_4_sim_not_activated
-  Scenario: Retrieves SIM swap date for a non-activated sim
+  @retrieve_sim_swap_date_4_sim_never_connected
+  Scenario: Retrieves SIM swap date for a sim that has never been connected to the network
```
Comment (gregory1g) [inline code/Test_definitions/sim-swap-retrieveDate.feature:49]: The test case looks a bit fuzzy for me. Sim was not connected to the network - ok, but was it replaced? One can purchase a SIM and request an replacement before it is connect it to the network. From the SimSwap definition (and potential attack vector) this is still a SimSwap.  

Suggestion: avoid "connected to the network" validation and keep things simple: Sim either was associated with a phone number (pre-provision, on first-call, different ways) or not. The date when this association is the SimSwap date, independent on what user did with that or other SIM cards associated with the phone number.

Comment (bigludo7) [reply code/Test_definitions/sim-swap-retrieveDate.feature:49]: I'm fine with the proposal to change wording from _connected_ to _associated_ if it works also for @fernandopradocabrillo

Comment (fernandopradocabrillo) [reply code/Test_definitions/sim-swap-retrieveDate.feature:49]: Yes I agree with the proposal. It is clearer and makes more sense that way, I'll adapt it
Thanks!

**Review by bigludo7 — APPROVED** [review]
LGTM

**Pull request merged**